src/Dialog/DialogBottomSheet.tsx

```diff
@@ -0,0 +1,327 @@
+import styled from 'styled-components'
+import {SxProp} from '../sx'
+import {useRefObjectAsForwardedRef} from '../hooks/useRefObjectAsForwardedRef'
+import Box from '../Box'
+import {get} from '../constants'
+
+const ANIMATION_DURATION = 300
+const FULL_HEIGHT = 95
+const HALF_HEIGHT = 50
+
+/**
+ * Props to customize the rendering of the Dialog.
+ */
+export interface DialogBottomSheetProps extends SxProp {
+  /**
+   * This method is invoked when a gesture to close the dialog is used
+   */
+  onClose: (gesture: 'close-button' | 'escape' | 'drag' | 'overlay') => void
+
+  /**
+   * Passed through to use as the draggable area of the dialog.
+   */
+  header: React.ReactNode
+
+  /**
+   * Default: "dialog". The ARIA role to assign to this dialog.
+   * @see https://www.w3.org/TR/wai-aria-practices-1.1/#dialog_modal
+   * @see https://www.w3.org/TR/wai-aria-practices-1.1/#alertdialog
+   */
+  role?: 'dialog' | 'alertdialog'
+
+  /**
+   * Identifies the element (or elements) that labels the element it is applied to.
+   */
+  ariaLabelledby: string
+
+  /**
+   * Identifies the element (or elements) that describes the element on which the attribute is set.
+   */
+  ariaDescribedby: string
+
+  /**
+   * Indicates whether an element is modal when displayed.
+   */
+  ariaModal: boolean
+}
+
+export default React.forwardRef<HTMLDivElement, PropsWithChildren<DialogBottomSheetProps>>((props, forwardedRef) => {
+  const {onClose, children, role, ariaLabelledby, ariaDescribedby, ariaModal, header, sx} = props
+
+  // 🔄 STATES
+
+  const [open, setIsOpen] = useState<boolean>(false)
+  const [snappedHeight, setSnappedHeight] = useState<number>(HALF_HEIGHT)
+  const [fireDelayedOnClose, setFireDelayedOnClose] = useState<
+    'close-button' | 'escape' | 'drag' | 'overlay' | undefined
+  >()
+
+  // 📎 REFERENCES
+
+  const dialogRef = useRef<HTMLDivElement>(null)
+  const startY = useRef(0)
+  const startHeight = useRef(0)
+  const isDragging = useRef(false)
+
+  useRefObjectAsForwardedRef(forwardedRef, dialogRef)
+
+  // 🧑‍🦽 ACCESSIBILITY
+
+  const isReduced = prefersReducedMotion()
+
+  // 🥊 ACTIONS
+
+  const showBottomSheet = () => {
+    setIsOpen(true)
+    updateSheetHeight(HALF_HEIGHT)
+  }
+
+  const hideBottomSheet = async (gesture: 'close-button' | 'escape' | 'drag' | 'overlay') => {
+    setIsOpen(false)
+    if (isReduced) {
+      onClose(gesture)
+    } else {
+      setFireDelayedOnClose(gesture)
+    }
+  }
+
+  const updateSheetHeight = (height: number) => {
+    if (!dialogRef.current) return
+    dialogRef.current.style.height = `${height}dvh`
+    setSnappedHeight(height)
+  }
+
+  // 🎪 EVENTS
+
+  const dragStop = (e: MouseEvent | TouchEvent) => {
+    if (!dialogRef.current) return
+
+    e.preventDefault()
+
+    isDragging.current = false
+    const sheetHeight = parseInt(dialogRef.current.style.height)
+
+    const isReduced = prefersReducedMotion()
```

Comment:
Non-blocking nitpick: `isReduced` should already be in scope and shouldn't need to be redefined. I don't think it would change between render and `onDragStop`.

Reply:
Whoops yes this is overkill. Fixed.